lib/core/src/Cardano/DB/Sqlite.hs

```diff
@@ -231,7 +233,8 @@ startSqliteBackend manualMigration migrateAll trace fp = do
         Left e -> do
             destroyDBLayer ctx
             pure $ Left e
-        Right _ -> pure $ Right ctx
+        Right ms -> do
+            pure $ Right (Quantity (length ms), ctx)
```

Comment:
Your assumption is correct: there can be manual migrations but no automatic ones. 
Still, the migration count here (and this particular fix) is solely about automatic ones. It comes as a work-around for persistent's behavior of deleting tables combined with the cascading deletion of foreign tables. 
For manual migration, we are in total control of what's being done and, it is very unlikely we would make a manual migration that requires rolling back the chain. If we ever do, then it'd be probably time to revisit this whole migration logic instead of patching up things on top of each other. 
The migration logic has become quite convoluted already.  Manual migrations shouldn't be included because they aren't causing implicit uncontrolled behavior here.

Reply:
@KtorZ wrote:

> For manual migration, we are in total control of what's being done and, it is very unlikely we would make a manual migration that requires rolling back the chain. If we ever do, then it'd be probably time to revisit this whole migration logic instead of patching up things on top of each other.

Fair enough! This is a convincing reason.